app/models/project.rb

```diff
@@ -91,6 +95,23 @@ def is_admin?(user)
     user && (admin_group.users.member?(user) || user.admin)
   end
 
+  def can_join?(user)
+    open_membership && !is_member?(user)
+  end
+
+  def can_leave?(user)
+    is_member?(user) && user.id != user_id
+  end
+
+  def check_box_disabled?(obs, user)
```

Comment:
Would you mind adding a comment to this?  I'm sure I'll eventually figure out what this means as I continue to read through the PR, but I'm sure I'm not the last person who will be confused when seeing this method out of context!